app/controllers/concerns/idempotentable.rb

```diff
@@ -0,0 +1,46 @@
+module Idempotentable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :validate_idempotency_key
```

Comment:
should those actions still live here if they are defined in the respective controllers now?

Reply:
Good spot - they definitely should not be there - now removed